### ISSUE TITLE
torch.compile causes segmentation fault on XPU with all nightly builds

### ISSUE THREAD

**Issue by xuhancn** [pytorch/pytorch#189341]
# GitHub Issue Draft: torch.compile Segfault on XPU

**Title**: `torch.compile` causes segmentation fault on XPU with all nightly builds

---

## 🐛 Bug Description

`torch.compile()` causes a segmentation fault on XPU with PyTorch nightly builds, regardless of backend or PyTorch version tested.

## Environment

- **PyTorch versions tested**: 
  - 2.14.0.dev20260707+xpu ❌
  - 2.14.0.dev20260630+xpu ❌  
  - 2.14.0.dev20260615+xpu ❌
- **Platform**: Intel XPU (1 device available)
- **Python**: 3.10
- **oneAPI**: 2025.3
- **OS**: Linux
- **Installation**: `pip install --pre torch pytorch-triton-xpu --index-url https://download.pytorch.org/whl/nightly/xpu`

**Key packages**:
- pytorch-triton-xpu: 3.6.0+git225cdbde
- triton-xpu: 3.7.2+git5fcc14d9
- intel-sycl-rt: 2026.0.0

### Minimal reproducer:

```python
import torch

def fn(x):
    return x * 2 + 1

x = torch.randn(4, 4, device='xpu')
print(f"Eager mode: {fn(x).shape}")  # ✅ Works

compiled_fn = torch.compile(fn, backend="inductor")
result = compiled_fn(x)  # ❌ Segmentation fault (core dumped)
```

### Steps:
1. Install PyTorch XPU nightly
2. Run the code above
3. Process crashes with segmentation fault

## Expected Behavior

torch.compile should successfully compile and execute the function on XPU, similar to CUDA/CPU behavior.

## Actual Behavior

```
Segmentation fault (core dumped)
Exit code: 139
```

- No Python exception raised
- Crash occurs during first invocation (JIT compilation phase)
- Affects ALL backends: `inductor`, `eager`, `aot_eager`

## What Works

✅ Basic XPU tensor operations  
✅ matmul, sigmoid, type conversions  
✅ Eager mode execution (without compile)  
✅ torch.xpu.is_available() returns True

## What Fails

❌ torch.compile with any backend  
❌ Even simplest functions (x * 2)  
❌ Consistent across multiple PyTorch nightly versions

## Additional Context

**Related Issue**: This blocks reproduction of #180948 (`test_emulate_precision_casts_preserves_explicit_precision_cast_float16`), which requires inductor compilation.

**Investigation**: Tested 3 different nightly versions spanning ~3 weeks. All exhibit the same crash, suggesting this is not a recent regression but a fundamental environmental or compatibility issue.

**Hypothesis**: Possible ABI incompatibility between pytorch-triton-xpu/triton-xpu and torch XPU build, or missing oneAPI configuration.

## Logs/Backtraces

(To be added with gdb backtrace if requested)

## CC

@chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv 

(XPU team members from #180948)

---

**Labels**: `module: xpu`, `module: inductor`, `triaged`, `oncall: pt2`

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo @gujinghui @fengyuan14 @guangyey

**Comment by xuhancn** [comment]
## Closing as Configuration Issue

After deep root cause analysis, this is **not a PyTorch bug** but an environment configuration issue.

### Root Cause

SYCL library version conflict between PyTorch and Triton-XPU:
- **PyTorch XPU** uses: `libsycl.so.9` (from conda package `intel-sycl-rt==2026.0.0`)
- **Triton-XPU** uses: `libsycl.so.8` when oneAPI `setvars.sh` is sourced

### Solution ✅

**Do NOT source oneAPI setvars.sh when using PyTorch XPU nightly conda packages:**

```bash
# ❌ Wrong
source ~/intel/oneapi/setvars.sh
conda activate env
python script.py  # Crashes

# ✅ Correct
conda activate env
python script.py  # Works perfectly
```

The conda environment provides all necessary SYCL dependencies.

### Technical Explanation

Triton-XPU's SYCL discovery logic prioritizes system oneAPI over conda packages. When oneAPI is sourced, `icpx` is in PATH, so Triton compiles against libsycl.so.8 while PyTorch uses libsycl.so.9 → ABI mismatch → segfault.

**Evidence from GDB:**
```
#0  libsycl.so.8: context_impl::get_info<devices>()  ← Crash
#1  libsycl.so.8: context::get_devices()
#2  spirv_utils.so: init_devices()
```

### Recommendation

This should be documented in PyTorch XPU installation guides to prevent user confusion.
